src/averageAges.js

```diff
@@ -14,14 +14,20 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+// - Here is some crazy solutions
+// |-- to solve it with no loops, if\else statements etc
+// |-- ...but a lot of variables))
+function calculateMenAverageAge(people, century = true) {
+  const men = people.filter(person => person.sex === 'm');
+  const ages = men
+    .map((person) =>
+      Math.ceil(person.died / 100) === century || century === true
+        ? person.died - person.born
+        : 0)
+    .filter(age => age !== 0);
```

Comment:
you can do all of these only with two filters
if we don't have century then filter only by sex, if we have century then filter by sex and Math.ceil

Something like this
```
ages = century
 ? filter by sex
 : filter be sex and math.ceil

return calc average
```